### PR TITLE
add authorization

### PR DESCRIPTION
**Pull request by Agetuni**
- install CanCan
- implement delete post and comment functionalities 
- Set Authorization for only admins to perform the task 

image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -12,4 +12,8 @@ class User < ApplicationRecord
   def recent_post
     posts.order('created_at Desc').limit(3)
   end
+
+  def admin?
+    role == 'admin'
+  end
```
Comment (OLIPLICHE) [inline app/models/user.rb:18]: - Nice Job for creating `user.admin?` method inside the user model 👏

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -29,4 +29,13 @@ def create
       end
     end
   end
+
```
Comment (OLIPLICHE) [inline app/controllers/posts_controller.rb:34]: - [ ] Please use **CanCanCan** for Post authorization in `Post Controller`:
- `load_and_authorize_resource` method is provided to automatically authorize all actions in a RESTful style resource controller. Here is the code: 
**class PostsController < ApplicationController
  load_and_authorize_resource**
- Open this Link [cancancan](https://github.com/CanCanCommunity/cancancan) and scroll down to Controller helper section, to Learn more

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**

- [ ]  To mention all the collaborators of this milestone in your ReadMe. Teamwork skills

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @Team, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**